components/director/pkg/log/formatter.go

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2020 The Compass Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package log
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+type formatterEntry struct {
+	WrittenAt        string        `json:"written_at"`
+	WrittenTimestamp string        `json:"written_ts"`
+	ComponentType    string        `json:"component_type"`
+	CorrelationID    string        `json:"correlation_id"`
+	Type             string        `json:"type"`
+	Logger           string        `json:"logger"`
+	Level            string        `json:"level"`
+	Message          string        `json:"msg"`
+	Fields           logrus.Fields `json:"-"`
+}
+
+// MarshalJSON marshals the kibana entry by inlining the logrus fields instead of being nested in the "Fields" tag
+func (k formatterEntry) MarshalJSON() ([]byte, error) {
+	type Entry formatterEntry
+	bytes, err := json.Marshal(Entry(k))
+	if err != nil {
+		return nil, err
+	}
+
+	var result map[string]json.RawMessage
+	if err = json.Unmarshal(bytes, &result); err != nil {
+		return nil, err
+	}
+
+	for k, v := range k.Fields {
+		field, err := json.Marshal(v)
+		if err != nil {
+			return nil, err
+		}
+		result[k] = field
+	}
+
+	return json.Marshal(result)
+}
+
+// Formatter is a logrus formatter that formats an entry for Kibana
+type Formatter struct {
```

Comment:
wondering if this formatter still brings value for the integration with the cloud logging kibana? have you tried how the log entries look with and without it ?

Reply:
In Cloud Logging Kibana we don't have any predefined dashboards, so it is up to us to define ones based on our preferred format. 
However, this format is established already and I think it is better to have it in the beginning in case we want to start fast with already familiar dashboards and enrich them in the future to match our preferred format.